### ISSUE TITLE
[DOC] contributing.md should point readers to relevant developer guides

### ISSUE THREAD

**Issue by harrism** [rapidsai/cudf#11348]
I was looking over the updated PR template (#10774), which refers developers to the contributing.md. But we have very detailed developer guides that contributors should probably read, and these are not mentioned by https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md.

e.g., for C++ contributors, the [libcudf Developer Guide](https://github.com/rapidsai/cudf/blob/branch-22.08/cpp/docs/DEVELOPER_GUIDE.md)

And for Python contributors, the [cuDF Developer Guide](https://github.com/rapidsai/cudf/tree/branch-22.08/docs/cudf/source/developer_guide)

Note there is a link to the latter document here that is broken: https://github.com/rapidsai/cudf/tree/branch-22.08/docs/cudf#readme

**Comment by davidwendt** [comment]
The broken link in the [readme](https://github.com/rapidsai/cudf/tree/branch-22.08/docs/cudf#readme) is to a new file and will be resolved once branch-22.08 is merged into the main branch.

**Comment by davidwendt** [comment]
> The broken link in the [readme](https://github.com/rapidsai/cudf/tree/branch-22.08/docs/cudf#readme) is to a new file and will be resolved once branch-22.08 is merged into the main branch.

I just learned that replacing the `/main/` path with `/HEAD/` in this URL will point to the tip of the default branch.
So, I will update that in #11390

**Comment by vyasr** [comment]
Resolved in #11390
